daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -568,6 +568,13 @@ convertExpr env0 e = do
             { retrieveByKeyTemplate = tmpl'
             , retrieveByKeyKey = EVar varV1
             }
+    go env (VarIs "unpackPair") (LType (StrLitTy f1) : LType (StrLitTy f2) : LType t1 : LType t2 : args)
```

Comment:
It looks like this should check the module name similar to the check for `Pair` below.

Reply:
Half our stuff should do that... In general any use of VarIs is wrong.